src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,537 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+import analysis.inner_product_space.spectrum
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `↥P _` for the extended orthogonal projection `V →L[ℂ] V`.
+
+We let `V` be an inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P` := orthogonal_projection
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def submodule.invariant (U : submodule ℂ V) (T : V →ₗ[ℂ] V) : Prop := U ≤ U.comap T
+lemma submodule.invariant_iff (U : submodule ℂ V) (T : V →ₗ[ℂ] V) :
+  submodule.invariant U T ↔ T '' U ⊆ U :=
+by simp only [set.image_subset_iff]; refl
+
+/-- `↥P _` is the extended version of `P _` -/
```

Comment:
The docstring should be understandable just by itself (it appears when one hovers any use of the theorem). Better phrase it with something like:
```suggestion
/-- The orthogonal projection on a submodule `U` of an inner product space `V`, as a continuous linear map from `V` to `V`. See also `orthogonal_projection` for the version as a continuous linear map from `V` to `U`.  -/
```